From 206dd13651bf473a9074c235e4a8eaa5528095ba Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:52 -0400
Subject: [virt] kvm/ppc: use fdget

Message-id: <1410765214-16377-37-git-send-email-dgibson@redhat.com>
Patchwork-id: 94755
O-Subject: [PATCH 36/78] powerpc kvm: use fdget
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Signed-off-by: David Gibson <dgibson@redhat.com>

commit 70abadedab73066fae584488640056f83bce5f95
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Fri Aug 30 15:04:22 2013 -0400

    powerpc kvm: use fdget

    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/powerpc.c b/arch/powerpc/kvm/powerpc.c
index 85df164..d3883ff 100644
--- a/arch/powerpc/kvm/powerpc.c
+++ b/arch/powerpc/kvm/powerpc.c
@@ -929,39 +929,39 @@ static int kvm_vcpu_ioctl_enable_cap(struct kvm_vcpu *vcpu,
 #endif
 #ifdef CONFIG_KVM_MPIC
  case KVM_CAP_IRQ_MPIC: {
-  struct file *filp;
+  struct fd f;
   struct kvm_device *dev;
 
   r = -EBADF;
-  filp = fget(cap->args[0]);
-  if (!filp)
+  f = fdget(cap->args[0]);
+  if (!f.file)
    break;
 
   r = -EPERM;
-  dev = kvm_device_from_filp(filp);
+  dev = kvm_device_from_filp(f.file);
   if (dev)
    r = kvmppc_mpic_connect_vcpu(dev, vcpu, cap->args[1]);
 
-  fput(filp);
+  fdput(f);
   break;
  }
 #endif
 #ifdef CONFIG_KVM_XICS
  case KVM_CAP_IRQ_XICS: {
-  struct file *filp;
+  struct fd f;
   struct kvm_device *dev;
 
   r = -EBADF;
-  filp = fget(cap->args[0]);
-  if (!filp)
+  f = fdget(cap->args[0]);
+  if (!f.file)
    break;
 
   r = -EPERM;
-  dev = kvm_device_from_filp(filp);
+  dev = kvm_device_from_filp(f.file);
   if (dev)
    r = kvmppc_xics_connect_vcpu(dev, vcpu, cap->args[1]);
 
-  fput(filp);
+  fdput(f);
   break;
  }
 #endif /* CONFIG_KVM_XICS */
-- 
1.7.1